lib/charms/zookeeper/v0/client.py

```diff
@@ -124,9 +125,10 @@ def __init__(
         self.keyfile_password = keyfile_password
         self.certfile_path = certfile_path
         self.leader = ""
+        self.read_only = read_only
 
         try:
-            self.leader = self.get_leader()
+            self.leader = self.get_leader(check_for_strict_leadership=not read_only)
```

Comment:
**nitpick/non-blocking:** I don't know why, but having `not X` in an argument value feels wrong.
Could we do:
```python
self.leader = self.get_leader() if not self.read_only else ""
```

Reply:
We cannot just go with your suggestion because we still need a host string assigned to `self.leader`.
How about renaming this attr to something else, like "zk_host", and conditionally use `get_leader` (with reverted changes) or a new method that basically picks the first of the `hosts` routing to a live ZK server?

I think it would be way simpler and more maintainable than my current suggestion